app/models/exam_registration.rb

```diff
@@ -22,8 +23,12 @@ def ensure_valid_authorization_criterion!
     authorization_criterion.ensure_valid!
   end
 
-  def start!(users)
-    users.each &method(:notify_user!)
+  def register!(users)
+    update! registrees: users
+  end
+
+  def start!
+    registrees.each &method(:notify_user!)
```

Comment:
I think that register should call start only for users that you are registering.
Now, if you start an exam registration and later add some users to the registration you will send the notification to all users, newer and olders.